vcsh

```diff
@@ -234,17 +235,20 @@ foreach() {
 	# We default to prefixing `git` to all commands passed to foreach, but
 	# allow running in general context with -g
 	command_prefix=git
-	while getopts "g" flag; do
+	command_suffix='cat'
+	while getopts "gp" flag; do
 		if [ x"$1" = x'-g' ]; then
 			unset command_prefix
 		fi
+		if [ x"$1" = x'-p' ]; then
```

Comment:
This method of option parsing just doesn't work right at all, you can't use `-g -p` or `-p -g` because the manual shift is eating the argument and throwing `getopts()` off. Speaking of which, this isn't how `getopts()` is supposed to work at all, and the flag variable it sets is being ignored. I don't know how the top level option parsing got away with this for so long, but I don't think it's correct either....

Reply:
See also #286 for a PR that fixes the `getopts()` usage for the main script.